core/src/main/java/io/lionweb/lioncore/java/serialization/data/SerializedContainmentValue.java

```diff
@@ -1,18 +1,22 @@
 package io.lionweb.lioncore.java.serialization.data;
 
+import java.util.ArrayList;
+import java.util.Collections;
 import java.util.List;
 import java.util.Objects;
 
 /** This represents the serialization of the values of a containment link in a Node. */
 public class SerializedContainmentValue {
   private MetaPointer metaPointer;
-  private List<String> value;
+  private final List<String> value;
 
-  public SerializedContainmentValue() {}
+  public SerializedContainmentValue() {
+    this.value = new ArrayList<>();
+  }
 
   public SerializedContainmentValue(MetaPointer metaPointer, List<String> value) {
     this.metaPointer = metaPointer;
-    this.value = value;
+    this.value = new ArrayList<>(value);
```

Comment:
Had to think why it would be useful to make a defensive copy here, but now I get it.

Reply:
Nobody shall touch my internal data structure directly.